frontend/src/views/Enrollment/EnrollmentForm.js

```diff
@@ -0,0 +1,310 @@
+import InterestsControl from '../../components/InterestsControl';
+import AvailabilitySelector from '../../components/AvailabilitySelector';
+import { participantType } from '../../data/participants';
+
+const useStyles = makeStyles(theme => ({
+  formCard: {
+    maxWidth: "80em",
+    margin: theme.spacing(2),
+  },
+  buttonWrapper: {
+    position: 'relative',
+  },
+  buttonProgress: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    marginTop: -12,
+    marginLeft: -12,
+  },
+}));
+
+const MENTORING_MANA_PAGE = "https://mana.mozilla.org/wiki/pages/viewpage.action?spaceKey=PR&title=Mozilla+Mentoring+Program";
+const CLG_PAGE = "https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=119683146";
+const ORG_CHART_DISTANCES = [
+  {key: "require-close", text: "Yes, someone close please"},
+  {key: "prefer-close", text: "I'd prefer someone close, but I'm flexible"},
+  {key: "neutral", text: "No preference"},
+  {key: "prefer-far", text: "I'd prefer someone far away, but I'm flexible"},
+  {key: "require-far", text: "No, someone far away please"},
+];
+const TRACK_CHANGE_INTEREST = [
+  {key: "yes", text: "Yes"},
+  {key: "maybe", text: "I'm thinking about it"},
+  {key: "no", text: "No"},
+];
+const TIMES_AT_ORG_LEVEL = [
+  {key: "<1 year", text: "Less than a year"},
+  {key: "1-2 years", text: "1-2 years"},
+  {key: "2-3 years", text: "2-3 years"},
+  {key: "3-5 years", text: "3-5 years"},
+  {key: ">6 years", text: "6 or more years"},
+];
+const CANNED_INTERESTS = [
+  "Technical Leadership",
+  "Public Speaking",
+  "Interpersonal Communication",
+  "Time Management",
+  "Increasing Impact on Mozilla's Mission",
+  "Getting Things Done at Mozilla (navigating organizational culture)",
+];
+
+// A link in a form control's help text, opening in a new page and not in the tab index.
+const InfoLink = ({ page, children }) => (
+  <a tabIndex="-1" rel="noreferrer" target="_blank" href={page}>{children}</a>
+);
+
+InfoLink.propTypes = {
+  page: propTypes.string.isRequired,
+  children: propTypes.node.isRequired,
+};
+
+export default function EnrollmentForm({ participant, update, onParticipantChange, onSubmit, submitLoading }) {
+  const classes = useStyles();
+
+  const mentor = participant.is_mentor;
+  const learner = participant.is_learner;
+  const either = mentor || learner;
+  const neither = !either;
+
+  const handleInputChange = event => {
+    const { name, value } = event.target;
+    onParticipantChange({
+      ...participant,
+      [name]: value,
+    })
+  };
+
+  const handleSwitchChange = event => {
+    const { name } = event.target;
+    onParticipantChange({
+      ...participant,
+      [name]: !participant[name],
+    });
+  };
+
+  const handleTimeAvailabilityChange = timeAvailability => {
+    onParticipantChange({
+      ...participant,
+      time_availability: timeAvailability,
+    });
+  }
+
+  const setLearnerInterests = learner_interests => {
```

Comment:
Nit: stick to camel case instead of alternating between snake case and camel case? :)